src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -5208,8 +5267,15 @@ def _make_onedeploy_request(params):
         response_body = None
         if poll_async_deployment_for_debugging:
             logger.info('Polling the status of async deployment')
+            if params.track_deployment.lower() == 'true':
+                # once deploymentstatus/latest is available, we can use it to track the deployment
+                deployment_id = _get_latest_deployment_id(params.cmd, params.resource_group_name, params.webapp_name, deployment_status_url, params.slot)
+                if deployment_id is None:
+                    raise CLIError("Unable to fetch deployment id for this deployment")
```

Comment:
if we can't find deployment_id, let's fallback to the old way instead of throwing error?

Reply:
But the user explicitly wants to use deploymentstatus endpoint, And we have a reasonable number of retries to get the id, wouldn't it be better to fail instead here?

Reply:
yeah, makes sense. Then, maybe we can update the error message to also ask them to try without `track_deployment` flag?